src/storage/store/engine.rs

```diff
@@ -308,6 +390,15 @@ impl ShardEngine {
         }
     }
 
+    pub fn sync_id_exists(&mut self, sync_id: &Vec<u8>) -> bool {
```

Comment:
Is this for test use, or public use? If test, I'd go with pub(crate).

Otherwise, probably best to surface the error back to the caller instead of logging and returning false. I think trie.exists should only fail under extreme conditions.

Reply:
Test, I'll fix